polkadot/xcm/xcm-builder/src/unique_instances/ops.rs

```diff
@@ -0,0 +1,154 @@
+/// using the [`FromTo`] strategy. Restoring with the [`IfRestorable`] is implemented symmetrically
+/// as the transfer from the stash account using the [`FromTo`] strategy.
+pub struct SimpleStash<StashAccount, InstanceOps>(PhantomData<(StashAccount, InstanceOps)>);
+impl<StashAccount, InstanceOps> AssetDefinition<Instance> for SimpleStash<StashAccount, InstanceOps>
+where
+	InstanceOps: AssetDefinition<Instance>,
+{
+	type Id = InstanceOps::Id;
+}
+impl<StashAccount, InstanceOps> Stash<Instance, IfOwnedBy<StashAccount::Type>>
+	for SimpleStash<StashAccount, InstanceOps>
+where
+	StashAccount: TypedGet,
+	InstanceOps: Transfer<Instance, FromTo<StashAccount::Type>>,
+{
+	fn stash(
+		id: &Self::Id,
+		IfOwnedBy(possible_owner): IfOwnedBy<StashAccount::Type>,
+	) -> DispatchResult {
+		InstanceOps::transfer(id, FromTo(possible_owner, StashAccount::get()))
+	}
+}
+impl<StashAccount, InstanceOps> Restore<Instance, IfRestorable<StashAccount::Type>>
+	for SimpleStash<StashAccount, InstanceOps>
+where
+	StashAccount: TypedGet,
+	InstanceOps: Transfer<Instance, FromTo<StashAccount::Type>>,
+{
+	fn restore(
+		id: &Self::Id,
+		IfRestorable(owner): IfRestorable<StashAccount::Type>,
+	) -> DispatchResult {
+		InstanceOps::transfer(id, FromTo(StashAccount::get(), owner))
+	}
+}
+
+/// The `RestoreOnCreate` implements the [`Create`] operation by utilizing the [`Restore`]
+/// operation. The creation is implemented using the [`Owned`] strategy with the [`PredefinedId`] ID
+/// assignment. Such creation is modeled by the [`Restore`] operation using the [`IfRestorable`]
+/// strategy.
+///
+/// The implemented [`Create`] operation can be used in the
+/// [`UniqueInstancesAdapter`](super::UniqueInstancesAdapter) via the [`UniqueInstancesOps`].
+pub struct RestoreOnCreate<InstanceOps>(PhantomData<InstanceOps>);
+impl<InstanceOps> AssetDefinition<Instance> for RestoreOnCreate<InstanceOps>
+where
+	InstanceOps: AssetDefinition<Instance>,
+{
+	type Id = InstanceOps::Id;
+}
+impl<AccountId, InstanceOps> Create<Instance, Owned<AccountId, PredefinedId<InstanceOps::Id>>>
+	for RestoreOnCreate<InstanceOps>
+where
+	InstanceOps: Restore<Instance, IfRestorable<AccountId>>,
+{
+	fn create(
+		strategy: Owned<AccountId, PredefinedId<InstanceOps::Id>>,
+	) -> Result<InstanceOps::Id, DispatchError> {
+		let Owned { owner, id_assignment, .. } = strategy;
+		let instance_id = id_assignment.params;
+
+		InstanceOps::restore(&instance_id, IfRestorable(owner))?;
+
+		Ok(instance_id)
+	}
+}
+
+/// The `StashOnDestroy` implements the [`Destroy`] operation by utilizing the [`Stash`] operation.
+/// The destroy operation is implemented using the [`IfOwnedBy`] strategy
+/// and  modeled by the [`Stash`] operation using the same strategy.
+///
+/// The implemented [`Destroy`] operation can be used in the
+/// [`UniqueInstancesAdapter`](super::UniqueInstancesAdapter) via the [`UniqueInstancesOps`].
+pub struct StashOnDestroy<InstanceOps>(PhantomData<InstanceOps>);
```

Comment:
I like the idea of implementing destroy using a stash instead of burning the asset. However, should stash be its own operation? I also have a hard time wrapping my head around strategies, some seem to make sense, but some seem like they're more implementation details

Reply:
I thought the `Stash` / `Restore` operations could be reasonably frequent. For instance, the fractionalization operation implies "stashing" the NFT and depositing associated fungibles (and symmetrically, burning fungibles and "restoring" the NFT).

The non-fungibles could be stashed during other, more high-level operations (like fractionalization). For instance, an NFT could be used as an entry ticket to a dedicated on-chain logic, and this ticket is considered "active" when the NFT is stashed. I feel that more use cases could be derived around stashing/restoring.

Nonetheless, I agree that these two strategies are more semantically complex than `Transfer` or `Destroy`.

Reply:
It could be. To me, the `Stash` seems more general than the `Lock`. Maybe I'm wrong, but it looks like the `Lock` should preserve ownership (the `Unlock` should return the token to the before-lock-owner). `Stash` isn't obliged to provide such guarantees; hence, it is more general.